temporal-test-server/src/main/java/io/temporal/internal/testservice/StateMachines.java

```diff
@@ -733,112 +729,60 @@ private static void scheduleNexusOperation(
-    }
-  }
-
-  private static void handleSyncSuccess(
-      RequestContext ctx, NexusOperationData data, StartOperationResponse.Sync response) {
+      RequestContext ctx, NexusOperationData data, Payload result, long notUsed) {
     ctx.addEvent(
         HistoryEvent.newBuilder()
             .setEventType(EventType.EVENT_TYPE_NEXUS_OPERATION_COMPLETED)
             .setNexusOperationCompletedEventAttributes(
                 NexusOperationCompletedEventAttributes.newBuilder()
                     .setRequestId(data.scheduledEvent.getRequestId())
                     .setScheduledEventId(data.scheduledEventId)
-                    .setResult(response.getPayload()))
+                    .setResult(result))
             .build());
   }
 
-  private static void handleUnsuccessfulOperationError(
-      RequestContext ctx, NexusOperationData data, UnsuccessfulOperationError err) {
-
-    Failure f =
-        Failure.newBuilder()
-            .setMessage("nexus operation completed unsuccessfully")
-            .setNexusOperationExecutionFailureInfo(
-                NexusOperationFailureInfo.newBuilder()
-                    .setEndpoint(data.endpoint.getSpec().getName())
-                    .setService(data.scheduledEvent.getService())
-                    .setOperation(data.scheduledEvent.getOperation())
-                    .setOperationId(data.operationId)
-                    .setScheduledEventId(data.scheduledEventId))
-            .setCause(nexusFailureToApplicationFailure(err.getFailure()))
-            .build();
-
-    HistoryEvent event;
-    if (data.cancelRequested) {
-      event =
-          HistoryEvent.newBuilder()
-              .setEventType(EventType.EVENT_TYPE_NEXUS_OPERATION_CANCELED)
-              .setNexusOperationCanceledEventAttributes(
-                  NexusOperationCanceledEventAttributes.newBuilder()
-                      .setRequestId(data.scheduledEvent.getRequestId())
-                      .setScheduledEventId(data.scheduledEventId)
-                      .setFailure(f))
-              .build();
-    } else {
-      event =
-          HistoryEvent.newBuilder()
-              .setEventType(EventType.EVENT_TYPE_NEXUS_OPERATION_FAILED)
-              .setNexusOperationFailedEventAttributes(
-                  NexusOperationFailedEventAttributes.newBuilder()
-                      .setRequestId(data.scheduledEvent.getRequestId())
-                      .setScheduledEventId(data.scheduledEventId)
-                      .setFailure(f))
-              .build();
-    }
-
-    ctx.addEvent(event);
-  }
-
   private static State timeoutNexusOperation(
       RequestContext ctx, NexusOperationData data, TimeoutType timeoutType, long notUsed) {
-    if (timeoutType != TimeoutType.TIMEOUT_TYPE_SCHEDULE_TO_CLOSE) {
+    if (timeoutType != TimeoutType.TIMEOUT_TYPE_SCHEDULE_TO_CLOSE
+        && timeoutType != TimeoutType.TIMEOUT_TYPE_SCHEDULE_TO_START) {
```

Comment:
Yeah, was using it to timeout an individual attempt, but will refactor this a bit since we are planning to add schedule to start support in the future.